rust/core-lib/src/styles.rs

```diff
@@ -220,4 +222,65 @@ impl ThemeStyleMap {
         self.styles.push(style.clone());
         result
     }
-}
+
+    /// Delete key from the themes map.
+    pub(crate) fn remove_theme(&mut self, path: &Path) -> Option<String> {
+        if let Some(theme_name) = path.file_stem().and_then(OsStr::to_str) {
+            self.themes.themes.remove(theme_name);
+            //Delete dump file
+            let _ = fs::remove_file(path.with_extension("tmdump"));
+            Some(theme_name.to_string())
+        } else { None }
+    }
+
+    /// Load all themes inside the given directory.
+    pub(crate) fn load_theme_dir(&mut self, themes_dir: PathBuf) {
+        match ThemeSet::discover_theme_paths(themes_dir) {
+            Ok(themes) => {
+                for theme in themes.iter() {
+                    self.load_theme(theme);
+                }
+            },
+            Err(e) => eprintln!("Error while loading themes dir: {:?}",e),
+        }
+    }
+
+    fn load_theme(&mut self, path: &Path) {
+        //Check for the dump file first
+        let theme_cache = path.with_extension("tmdump");
```

Comment:
A minor possible issue with this (I believe) is that the cache could be stale if the theme file was modified while the editor was closed.

One possible solution would be to associate the cache file with the theme file's modification time?